components/script/dom/audioparam.rs

```diff
@@ -106,6 +112,22 @@ impl AudioParamMethods for AudioParam {
 
     // https://webaudio.github.io/web-audio-api/#dom-audioparam-automationrate
     fn SetAutomationRate(&self, automation_rate: AutomationRate) {
+        if automation_rate == AutomationRate::A_rate {
+            if self.node_type == AudioNodeType::AudioListenerNode {
+                // ignore
+                return;
+            }
+
+            if self.node_type == AudioNodeType::AudioBufferSourceNode &&
+                (self.param == ParamType::Detune || self.param == ParamType::PlaybackRate)
+            {
+                // TODO: this should raise InvalidState
```

Comment:
mark the attribute as `[SetterThrows]` in webidl, make this function return `Fallible<()>`, and return `Err(Error::InvalidState)`